### ISSUE TITLE
x/playground: error line highlighting is broken

### ISSUE THREAD

**Issue by adg** [golang/go#17187]
https://play.golang.org/p/hO3Iguuryi

When you click 'run' line 8 should be highlighted as a syntax error, but this appears to be broken. I suspect it's because the regexp in the frontend JS isn't matching the error message properly.

```
tmp/sandbox073288310/main.go:8: undefined: asf
```

The sandbox server should probably trim the directory from the error message, too. In fact, that might be what broke it.

**Comment by fredr** [comment]
The regexp seems to be wrong indeed 
```
var regex = /prog.go:([0-9]+)/g;
``` 

But the `playground.js` doesn't seem to be part of https://go.googlesource.com/playground/ or am I blind? 👀

**Comment by ALTree** [comment]
I believe `playground.js` is imported from here: https://github.com/golang/tools/blob/master/godoc/static/playground.js

**Comment by fredr** [comment]
All right! thanks @ALTree. The highlighting works on https://tour.golang.org/ though. So the issue might be something else. But I'll have a look whenever I have some time over.

**Comment by fredr** [comment]
Oops, never got around to fix this, but I believe this will be fixed when #23201 is fixed
